### DISCUSSION TITLE
scrollable snap for visible item

### DISCUSSION THREAD

**Discussion by oknozor** [hecrj/iced#936] [Q&A]
Hello, I am trying to implement the following scroll behaviour :
I have a scrollable widget containing rows wich look like this : 
![Screenshot from 2021-06-27 14-20-37](https://user-images.githubusercontent.com/24563836/123544301-f1443000-d752-11eb-8bab-b6530b2ad915.png)

I'd like it to auto scroll so the last selected element is always visible (the user is updating the selection using arrow keys). 
![Screenshot from 2021-06-27 14-22-53](https://user-images.githubusercontent.com/24563836/123544436-8d6e3700-d753-11eb-9a8e-d5a4be365ef8.png)

I am not sure if that is possible with the `snap_to` function since I need to be able to get the height  of the visible part of the scrollable widget to calculate an offset and the widow is resizable. 

Is there a way to get the current height of a widget ?

**Comment by 90-008** [comment]
I think you should be able to calculate the offset by keeping which item is focused (from the top), how many items are in the scrollable, then calculate the offset using `(1.0 / total_items) * current_item`.
